File: frontend/heartmage-candles/src/typesV2/order/ConfiguredCandle.ts
import { Image } from '../shared/BaseProduct';

export interface ConfiguredCandle {
  candle: Candle;
  decor: Decor;
  layerColors?: LayerColor[];
  numberOfLayer: NumberOfLayer;
  smell?: Smell;
  wick: Wick;
}

interface Candle {
  id: number;
  title: string;
  price: number;
  weightGrams: number;
  images: Image[];
}

interface Decor {
  id: number;
  title: string;
  price: number;
}

interface LayerColor {
  id: number;
  title: string;
  pricePerGram: number;
}

interface Smell {
  id: number;
  title: string;
  price: number;
}

interface Wick {
  id: number;
  title: string;
  price: number;
}

interface NumberOfLayer {
  id: number;
  number: number;
}

File: backend/HeartmadeCandles/HeartmadeCandles.Order.Core/Models/ConfiguredCandle.cs
namespace HeartmadeCandles.Order.Core.Models;

public class ConfiguredCandle
{
    public required Candle Candle { get; init; }
    
    public Decor? Decor { get; init; }
    
    public required LayerColor[] LayerColors { get; init; }
    
    public required NumberOfLayer NumberOfLayer { get; init; }
    
    public Smell? Smell { get; init; }
    
    public required Wick Wick { get; init; }
}
